src/snmalloc/pal/pal_windows.h

```diff
@@ -101,15 +101,31 @@ namespace snmalloc
       return EXCEPTION_CONTINUE_EXECUTION;
     }
 
+    // Keep a handle for the exception handler, so we can remove it later
+    // when needed.
+    inline static PVOID g_Handler{};
+
     static void initialise_for_singleton(size_t*) noexcept
     {
-      AddVectoredExceptionHandler(1, HandleReadonlyLazyCommit);
+      if (!g_Handler)
+      {
+        g_Handler = AddVectoredExceptionHandler(1, HandleReadonlyLazyCommit);
+      }
+    }
```

Comment:
How about something like this:

```suggestion
      // Keep a handle for the exception handler, so we can remove it later
      // when needed.
      static PVOID g_Handler{};

      // Destructor for removing exception handler.
      static OnDestruct tidy([](){
        if (g_Handler)
        {
          RemoveVectoredExceptionHandler(g_Handler);
          g_Handler = NULL; // Prevent dangling pointer
        }
      });
      // Add exception handler for lazy commit.
      if (!g_Handler)
      {
        g_Handler = AddVectoredExceptionHandler(1, HandleReadonlyLazyCommit);
      }
    }
```

It wouldn't need the Singleton to change, and I believe should get called in the right place to remove the exception handler?

Reply:
This is using the 
 
https://github.com/microsoft/snmalloc/blob/a3fe420af250f4662aced78b167e45166d202898/src/snmalloc/ds_core/helpers.h#L71-L86

which is how I typically get things to get destructed.

Reply:
That seems like it was made exactly for this situation 👍 

I'll just try it out today and update the PR.